integration-tests/test_connect.py

```diff
@@ -527,3 +527,348 @@ def get_template_repos_by_name(template_name, username, password, template_url):
             print(f"Response status: {response.status_code}")
             print(f"Response content (first 200 chars): {response.text[:200]}...")
         raise
+
+
+@pytest.mark.tier1
+@pytest.mark.parametrize("feature", ["remote-management", "content", "analytics"])
+def test_connect_with_single_feature_disabled(external_candlepin, rhc, test_config, feature):
+    """
+    :id: 4168e32d-f70e-47db-8aa1-49fcb730e17c
+    :title: Verify RHC connection with one feature disabled using --disable-feature flag
+    :parametrized: yes
+    :description:
+        This test verifies that when connecting to RHC with one feature explicitly
+        disabled using the --disable-feature flag, the expected features are disabled
+        based on dependency rules. When content or analytics is disabled, remote_management
+        is also automatically disabled as a dependency. The test validates this by
+        examining the JSON output of the connect command.
+    :tags: Tier 1
+    :steps:
+        1.  Ensure the system is disconnected from RHC.
+        2.  Run the 'rhc connect' command with --disable-feature flag for the specified feature and JSON output format.
+        3.  Verify that RHC reports being registered.
+        4.  Verify that the yggdrasil service is inactive.
+        5.  Parse the JSON output and verify the disabled feature shows enabled=False.
+        6.  Verify that if content or analytics is disabled, remote_management is also disabled.
+        7.  Verify that other features show the correct enabled status.
+    :expectedresults:
+        1.  The system is successfully disconnected (if previously connected).
+        2.  The 'rhc connect' command executes without error.
+        3.  RHC indicates the system is registered.
+        4.  The yggdrasil service is not in an active state.
+        5.  The explicitly disabled feature has enabled=False in the JSON output.
+        6.  If content or analytics is disabled, remote_management also has enabled=False.
+        7.  Other features show the correct enabled status based on dependency rules.
+    """
+    with contextlib.suppress(Exception):
+        rhc.disconnect()
+
+    command_args = prepare_args_for_connect(test_config, auth="activation-key", output_format="json")
+
+    command_args.extend(["--disable-feature", feature])
+
+    command = ["connect"] + command_args
+
+    result = rhc.run(*command)
+
+    assert rhc.is_registered
+    assert not yggdrasil_service_is_active()
+
+    json_output = json.loads(result.stdout)
+    features = json_output["features"]
+
+    # Map CLI feature names to JSON feature names
+    feature_mapping = {
+        "remote-management": "remote_management",
```

Comment:
Oh, it is pity that we use `remote-management` on the CLi and `remote_management` in the JSON output. I am sorry for that. We cannot do anything about this now due to backward compatibility.